core/pyos.py

```diff
@@ -294,109 +315,43 @@ def Sigaction(sig_num, handler):
+  # type: (int) -> None
+  """Have the kernel notify the main loop about the given signal"""
+  global gSignalHandler
+  assert gSignalHandler is not None
+  signal.signal(sig_num, gSignalHandler)
 
-  def __init__(self):
-    # type: () -> None
-    self.sigwinch_handler = None  # type: SigwinchHandler
-    self.last_sig_num = 0  # MUTABLE GLOBAL, for interrupted 'wait'
-    # signal/hook name -> handler
-    self.hooks = {}  # type: Dict[str, _TrapHandler]
-    self.traps = {}  # type: Dict[int, _TrapHandler]
-    # appended to by signal handlers
-    self.nodes_to_run = []  # type: List[command_t]
-
-  def InitShell(self):
-    # type: () -> None
-    """Always called when initializing the shell process."""
-    pass
-
-  def InitInteractiveShell(self, display, my_pid):
-    # type: (_IDisplay, int) -> None
-    """Called when initializing an interactive shell."""
-    # The shell itself should ignore Ctrl-\.
-    signal.signal(signal.SIGQUIT, signal.SIG_IGN)
-
-    # This prevents Ctrl-Z from suspending OSH in interactive mode.
-    signal.signal(signal.SIGTSTP, signal.SIG_IGN)
-
-    # More signals from
-    # https://www.gnu.org/software/libc/manual/html_node/Initializing-the-Shell.html
-    # (but not SIGCHLD)
-    signal.signal(signal.SIGTTOU, signal.SIG_IGN)
-    signal.signal(signal.SIGTTIN, signal.SIG_IGN)
-
-    # Register a callback to receive terminal width changes.
-    # NOTE: In line_input.c, we turned off rl_catch_sigwinch.
-
-    # This is ALWAYS on, which means that it can cause EINTR, and wait() and
-    # read() have to handle it
-    self.sigwinch_handler = SigwinchHandler(display, self)
-    signal.signal(signal.SIGWINCH, self.sigwinch_handler)
-
-    # This doesn't make any tests pass, and we might punt on job control
-    if 0:
-      try:
-        # Put the interactive shell in its own process group, named by its PID
-        posix.setpgid(my_pid, my_pid)
-        # Attach the terminal (stdin) to the progress group
-        posix.tcsetpgrp(0, my_pid)
-
-      except (IOError, OSError) as e:
-        # For some reason setpgid() fails with Operation Not Permitted (EPERM) under pexpect?
-        pass
-
-  def GetLastSignal(self):
-    # type: () -> int
-    """Return the last signal that fired"""
-    return self.last_sig_num
-
-  def AddUserTrap(self, sig_num, handler):
-    # type: (int, Any) -> None
-    """For user-defined handlers registered with the 'trap' builtin."""
 
-    if sig_num == signal.SIGWINCH:
-      assert self.sigwinch_handler is not None
-      self.sigwinch_handler.user_handler = handler
-    else:
-      signal.signal(sig_num, handler)
-    self.traps[sig_num] = handler
-    # TODO: SIGINT is similar: set a flag, then optionally call user _TrapHandler
+def GetPendingSignals():
+  # type: () -> List[int]
+  """Transfer ownership of the current queue of pending signals to the caller."""
+  global gSignalHandler
+  assert gSignalHandler is not None
+  return gSignalHandler.TakeSignalQueue()
```

Comment:
Why not use just one name?  e.g. EITHER GetPendingSignals or TakeSignalQueue (I like the latter)

Reply:
I like the latter too. Not sure why I gave these separate names...